Bunkum.Protocols.Https/SslConfiguration.cs

```diff
@@ -0,0 +1,15 @@
+using Bunkum.Core.Configuration;
+
+namespace Bunkum.Protocols.Https;
+
+public class SslConfiguration : Config
+{
+    public override int CurrentConfigVersion => 1;
+    public override int Version { get; set; }
+    
+    protected override void Migrate(int oldVer, dynamic oldConfig)
+    {}
+
+    public string SslCertificate { get; set; } = "cert.pfx";
+    public string? CertificatePassword { get; set; } = "password here or null";
+}
```

Comment:
This is an exact duplicate of the one in Gemini.

Should probably have TLS stuff in a project (e.g. `Bunkum.Protocols.TlsSupport` or something) that Https and Gemini can include from for convenience.

Could also help reduce some duplicate code between `SocketGeminiListener` and `SocketHttpsListener` as well, since those also have fairly similar codepaths. Not to mention the `BunkumServer` components as well.

This would also ease implementation of new protocols that require TLS.

Reply:
it originally wasnt a duplicate of the one in gemini, but it now is after some refactoring, i'll merge them